### PR TITLE
Improving the XML comments around device code flow

### PR DESCRIPTION
**Pull request by jmprieur**
Improving the XML comments around device code flow
Also removing the explanation for the empty PublicClientApplication on .NET core as we now have all the flows

### CODE REVIEW

File: msal/src/Microsoft.Identity.Client/Features/DeviceCode/DeviceCodeResult.cs [hunk 1]
```diff
@@ -1,29 +1,29 @@
-﻿//------------------------------------------------------------------------------
```
Comment (jmprieur) [inline msal/src/Microsoft.Identity.Client/Features/DeviceCode/DeviceCodeResult.cs:1]: I've no idea what happened to the heading comments here ?  I did not change anything there?

File: msal/src/Microsoft.Identity.Client/PublicClientApplication.cs [hunk 2]
```diff
@@ -1,496 +1,494 @@
...
-using System;
-using System.Threading.Tasks;
-using Microsoft.Identity.Client.Internal;
-using Microsoft.Identity.Client.Internal.Requests;
-using System.Collections.Generic;
-using Microsoft.Identity.Core;
-using Microsoft.Identity.Core.Instance;
-using Microsoft.Identity.Core.UI;
-using Microsoft.Identity.Core.Telemetry;
+﻿//------------------------------------------------------------------------------
```
Comment (MarkZuber) [inline msal/src/Microsoft.Identity.Client/PublicClientApplication.cs:1]: do we have a whitespace or line ending difficulty here?  this is weird that it's diffing the whole file for what appears to be a few lines...

File: msal/src/Microsoft.Identity.Client/Features/DeviceCode/IPublicClientApplication.cs [hunk 3]
```diff
@@ -1,102 +1,132 @@
...
         /// <summary>
-        /// Acquires device code from the authority and returns it to the caller via
-        /// the deviceCodeResultCallback. The function then proceeds to poll for the security
-        /// token which is granted upon successful login based on the device code information.
+        /// Acquires a security token on a device without a Web browser, by letting the user authenticate on 
+        /// another device. This is done in two steps:
+        /// <list type="bullet">
+        /// <item><description>the method first acquires a device code from the authority and returns it to the caller via
+        /// the <paramref name="deviceCodeResultCallback"/>. This callback takes care of interacting with the user
+        /// to direct him to authenticate (to a specific URL, with a code)</description></item>
```
Comment (MarkZuber) [inline msal/src/Microsoft.Identity.Client/Features/DeviceCode/IPublicClientApplication.cs]: nit:  should we have a gender-neutral stance on pronouns?  "to direct them" perhaps?

**Review by MarkZuber — APPROVED** [review]
A few minor nits, but content looks good.

File: msal/src/Microsoft.Identity.Client/PublicClientApplication.cs (hunk 2 above)
Comment (jmprieur) [reply msal/src/Microsoft.Identity.Client/PublicClientApplication.cs:1]: yes, I know. I don't understand what's going on for this file (and for the first comments of the other files). Is my VS doing something weird. that's the first time this happens.

File: msal/src/Microsoft.Identity.Client/Features/DeviceCode/IPublicClientApplication.cs (hunk 3 above)
Comment (jmprieur) [reply msal/src/Microsoft.Identity.Client/Features/DeviceCode/IPublicClientApplication.cs]: thanks. Done!

**Review by MarkZuber — APPROVED** [review]

**Pull request merged**